### PR TITLE
Fix for ensure Table Block Delete button displays when creating an auto generated screen.

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description

Fixes an issue where config required to display the `Delete` button in the table block is missing initially. 
- Suitable label defaults are now added when generating auto screens
- In the event that these are missing, the delete button will remain visible until the `Delete button` field is explicitly cleared 

Addresses: 
- https://github.com/Budibase/budibase/issues/11774

### CODE REVIEW

File: packages/client/src/components/app/blocks/TableBlock.svelte [hunk 1]
```diff
@@ -249,7 +249,7 @@
             props={{
               dataSource,
               saveButtonLabel: sidePanelSaveLabel || "Save", //always show
-              deleteButtonLabel: deleteLabel, //respect config
+              deleteButtonLabel: deleteLabel == "" ? "" : "Delete",
```
Comment (Ghrehh) [inline packages/client/src/components/app/blocks/TableBlock.svelte:252]: Does the non-strict equality get us anything an impilict truthiness check wouldn't here? Just checking as `==` always confuses me 🙃

**Review by Ghrehh — APPROVED** [review]
LGTM but small question

File: packages/client/src/components/app/blocks/TableBlock.svelte (hunk 1 above)
Comment (deanhannigan) [reply packages/client/src/components/app/blocks/TableBlock.svelte:252]: Yep, there are two things in play here. There's a backwards compatibility check at the top that looks for the old toggle boolean and ensures that it is taken into account first.

If the legacy value was explicitly `false` we change the label value to `""`. This implies then, on line 252, that a user decision to hide the button was made and so an empty string will now be used for the `Delete button` label, hiding it.

If no value was ever set for the `Delete button` label in the screen config, this value is explicitly `undefined` and considered to have never been set. In that scenario the default behaviour is to show the button with the `Delete` label.

Comment (deanhannigan) [reply packages/client/src/components/app/blocks/TableBlock.svelte]: I had a look over this area and decided to refactor all the delete label code into its own function. It was difficult to follow and the line you highlighted is actually a mistake on my part. Cheers 👍

**Review by aptkingston — APPROVED** [review]

**Pull request merged**